tests/platform_tests/fwutil/conftest.py

```diff
@@ -0,0 +1,91 @@
+import tarfile
+import json
+import pytest
+
+from random import randrange
+
+from fwutil_common import show_firmware
+
+DUT_HOME="/home/admin"
+DEVICES_PATH="/usr/share/sonic/device"
+FS_PATH_TEMPLATE = "/host/image-{}/fs.squashfs"
+FS_RW_TEMPLATE = "/host/image-{}/rw"
+FS_WORK_TEMPLATE = "/host/image-{}/work"
+FS_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-fs"
+OVERLAY_MOUNTPOINT_TEMPLATE = "/tmp/image-{}-overlay"
+
+def pytest_generate_tests(metafunc):
+    val = metafunc.config.getoption('--fw-pkg')
+    if 'fw_pkg_name' in metafunc.fixturenames and val is not None:
+        metafunc.parametrize('fw_pkg_name', [val], scope="module")
+
+@pytest.fixture(scope='module')
+def fw_pkg(fw_pkg_name):
+    with tarfile.open(fw_pkg_name, "r:gz") as f:
+        f.extractall()
+        with open('firmware.json', 'r') as fw:
+            fw_data = json.load(fw)
+            yield fw_data
+        for m in f.getmembers():
```

Comment:
@alexrallen why not to extract to tmp dir and then simply remove that dir?

Reply:
Yes, this would be better, thanks for the suggestion.